### PR TITLE
fix: es6 syntax in client

### LINKED ISSUE #1981: [critical] webpack-dev-server broken in old browsers
- Operating System: no matter
- Node Version: no matter
- NPM Version: no matter
- webpack Version: no matter
- webpack-dev-server Version: 3.6.0

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code

Ref https://github.com/webpack/webpack-dev-server/pull/1933

### Expected Behavior

No ES6 syntax in client

### Actual Behavior

Exists `class`

### For Bugs; How can we reproduce the behavior?

1. Clone repo
2. Create src/index.js (any content) in root project directory
3. Create index.html with `<script src="./main.js"></script>` in `root` project directory
4. Run `npm run prepare && node bin/webpack-dev-server.js`
5. Open http://localhost:8080/main.js as source and search `class BaseClient`

/cc @hiroppy @Loonride 

Any ideas?

### PR DESCRIPTION
**Pull request by evilebottnawi**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

no, it is critical fix, test will be in future

### Motivation / Use-Case

fixes #1981 

### Breaking Changes

No

### Additional Info

/cc @hiroppy @Loonride I am temporary block merge any PRs before we refactor tests/tests strcuture/structure inside lib (for any branches)

Please don't touch next branch before we do this in master

### CODE REVIEW

**Comment by evilebottnawi** [comment]
Test structure:
```
- test
  - e2e
  - cli
  - client
    - clients
  - server
    - utils
```

Test naming:
1. class tests have same name as class -> `MyClass.js` -> `MyClass.test.js`
2. pure file like `defaultPort.js` -> `defaultPort.test.js`
3. options for server should be placed in `server` directory and have name `compress-option.test.js`
4. options for client should be placed in `client` directory and have name `sockPath-options.test.js`

Next we should improve test performance, many tests are very long and written in effective way.

I am WIP on this.

File: test/e2e/ProvidePlugin.test.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const testServer = require('../helpers/test-server');
 const config = require('../fixtures/provide-plugin-config/webpack.config');
 const runBrowser = require('../helpers/run-browser');
 
-describe('ProvidePlugin', () => {
+describe.skip('ProvidePlugin', () => {
```
Comment (evilebottnawi) [inline test/e2e/ProvidePlugin.test.js]: Need rewrite tests

**Comment by evilebottnawi** [comment]
Merge without review because it is critical, feel free to feedback

**Comment by hiroppy** [comment]
I would like to separate utils tests. Utils shouldn't have to test through the server. It is e2e.

**Comment by evilebottnawi** [comment]
@hiroppy yes, utils should have own unit tests and in `Server.test.js` we should do integration test (example - check when we run server configuration has HotModuleReplacmentPlugin and etc)

**Comment by evilebottnawi** [comment]
@hiroppy https://github.com/webpack/webpack-dev-server/issues/1983

**Pull request merged**